src/App.js

```diff
@@ -0,0 +1,169 @@
+import { TodoItem, todoItemTemplate } from "./components/TodoItem.js";
+
+var todoInput = document.getElementById('new-todo-title');
+var todoList = document.getElementById('todo-list');
+var todoCount = document.querySelector('strong');
+
+var btn_all = document.querySelector('.all');
+var btn_active = document.querySelector('.active');
+var btn_completed = document.querySelector('.completed');
+
+var todoItems = new TodoItem(JSON.parse(localStorage.getItem('todoItems')));
+
+var todoItemID = (todoItems !== undefined && todoItems !== null) ? todoItems.length : 0;
+
+window.onload = () => {
+    todoInput.addEventListener('keydown', onChangeTodoInput);
+
+    btn_all.addEventListener('click', showAllTodoList);
+    btn_active.addEventListener('click', showActiveTodoList);
+    btn_completed.addEventListener('click', showCompletedTodoList);
+
+    render(todoItems);
+}
+
+const addTodoItem = (contents) => {
+    let item = {
+        id : todoItemID++,
+        status : "view",
+        contents : contents
+    }
+    
+    todoItems.push(item);
+    render(todoItems);
+}
+
+const onChangeTodoInput = (e) => {    
+    if (e.keyCode === 13) {
+        addTodoItem(e.target.value);
+        e.target.value = "";
+    }
+}
+
+const render = (items) => {
+    let template = items.map(item => todoItemTemplate(item));
+    todoList.innerHTML = template.join("");
+
+    todoList.addEventListener('click', onToggleTodoItem);
+    todoList.addEventListener('dblclick', onModifyTodoItem);
+    todoList.addEventListener('keydown', onChangeTodoItem)
+    todoList.addEventListener('click', onDestroyTodoItem);
+    
+    todoCount.innerHTML = todoItems.length;
+
+    localStorage.setItem('todoItems', JSON.stringify(todoItems));
+}
+
+const showAllTodoList = () => {
+    btn_all.classList.add("selected");
+    btn_active.classList.remove("selected");
+    btn_completed.classList.remove("selected");
+
+    render(todoItems);
+}
+
+const showActiveTodoList = () => {
+    btn_all.classList.remove("selected");
+    btn_active.classList.add("selected");
+    btn_completed.classList.remove("selected");
+
+    render(todoItems.filter(item => item.status !== "completed"));
+}
+
+const showCompletedTodoList = () => {
+    btn_all.classList.remove("selected");
+    btn_active.classList.remove("selected");
+    btn_completed.classList.add("selected");
+
+    render(todoItems.filter(item => item.status === "completed"));
+}
+
+const onToggleTodoItem = (e) => {
+    if (e.target.className === "toggle") {
+        todoItems = todoItems.map(
+            item => {
+                if (item.id == e.target.id) {
+                    return item = {
+                        ...item,
+                        status : "completed"
+                    }
+                } else {
+                    return item;
+                }
+            }
+        );
```

Comment:
```suggestion
    if (e.target.className !== "toggle") {
        return;
    }
    todoItems = todoItems.map(
        item => {
            if (item.id !== e.target.id) {
                return item;
            }
            return item = {
                ...item,
                status : "completed"
            }
        }
    );
```

if 중첩문을 제거하면 가독성이 올라갈 수 있어요. 충족할 수 없는 조건일 경우를 먼저 처리하고, 실제로 처리해야 하는 로직을 가장 마지막에 처리하도록 변경해보는건 어떨까요? 아래 코드에서 중첩 if문도 모두 개선해 볼 수 있을 것 같아요!

Reply:
감사해요! if 중첩을 최대한 줄여보는 습관을 가져보겠습니다 👼